### PR TITLE
Use ITK_INSTALL_*_DIR paths for standalone installation

### LINKED ISSUE #86: Installation path of header files
When building and installing this MetaIO library stand-alone, with `CMAKE_INSTALL_PREFIX=/opt/metaio` on macOS, the installed files are:

```
/opt/metaio
├── include
│   ├── itkMetaIO
│   │   ├── itkThirdParty.h
│   │   ├── itk_zlib.h
│   │   └── metaIOConfig.h
│   ├── localMetaConfiguration.h
│   ├── metaArray.h
│   ├── metaArrow.h
│   ├── metaBlob.h
│   ├── metaCommand.h
│   ├── metaContour.h
│   ├── metaDTITube.h
│   ├── metaEllipse.h
│   ├── metaEvent.h
│   ├── metaFEMObject.h
│   ├── metaForm.h
│   ├── metaGaussian.h
│   ├── metaGroup.h
│   ├── metaIOConfig.h
│   ├── metaITKUtils.h
│   ├── metaImage.h
│   ├── metaImageTypes.h
│   ├── metaImageUtils.h
│   ├── metaLandmark.h
│   ├── metaLine.h
│   ├── metaMesh.h
│   ├── metaObject.h
│   ├── metaOutput.h
│   ├── metaScene.h
│   ├── metaSurface.h
│   ├── metaTransform.h
│   ├── metaTube.h
│   ├── metaTubeGraph.h
│   ├── metaTypes.h
│   ├── metaUtils.h
│   └── metaVesselTube.h
└── lib
    └── libITKMetaIO.a
```

The `localMetaConfiguration.h` contains the statement `#include "itk_zlib.h"`, but this header file is in subdirectory `itkMetaIO`, requiring the addition of this include path to the list of include search paths. It seems to me that the install paths of the header files are inconsistent.

For comparison, when build and installed as part of ITK with `CMAKE_INSTALL_PREFIX=/opt/itk`, all header files of MetaIO end up in `/opt/itk/include/ITK-$major.$minor/`. Given this, I would expect that all header files of the stand-alone MetaIO library would end up in `/opt/metaio/include/itkMetaIO` instead of the top-level include directory.

Alternatively, the `itkMetaIO` subdirectory could be dropped from the stand-alone installation, given the `itk` file name prefixes and redundant `itkMetaIO/metaIOConfig.h`?

### PR DESCRIPTION
**Pull request by schuhschuh**
One proposed fix for #86, also affecting `METAIO_INSTALL_(LIB|ARCHIVE)_DIR`.

Closes #86.

### CODE REVIEW

**Comment by bradking** [comment]
Where are the `METAIO_INSTALL_*` variables referenced?  All the install rules are already using `ITK_INSTALL_*`.

**Comment by schuhschuh** [comment]
These variables are defined in `src/CMakeLists.txt` of this project at:

https://github.com/Kitware/MetaIO/blob/9b37f8e4b453e2a0eb3ec0ae2b9bf2d63d933514/src/CMakeLists.txt#L36

And used in installation rules at:

https://github.com/Kitware/MetaIO/blob/9b37f8e4b453e2a0eb3ec0ae2b9bf2d63d933514/src/CMakeLists.txt#L159

The top-level `CMakeLists.txt` basically would replace the role of the respective CMakeLists files used in ITK and VTK where the third-party library is copied in. See [this file](https://github.com/Kitware/ITK/blob/725689ba2aec77a7d379517dd559fe8399e2c007/Modules/ThirdParty/MetaIO/src/CMakeLists.txt) for ITK.

**Comment by schuhschuh** [comment]
For stand-alone build, should one point CMake to the `src/` subdirectory instead of the top-level?

**Comment by schuhschuh** [comment]
Otherwise, I would probably also add the following line to the top-level CMakeLists:

```
SET(METAIO_INSTALL_EXPORT_NAME ${METAIO_NAMESPACE})
```

where `METAIO_NAMESPACE` is `ITKMetaIO` instead of `MetaIO` which is set by `src/CMakeLists.txt`.

**Comment by schuhschuh** [comment]
> For stand-alone build, should one point CMake to the `src/` subdirectory instead of the top-level?

This doesn't work, as then zlib (specifically `itk_zlib.h`) is not found. Also would need to add a `cmake_minimum_required` to `src/CMakeLists.txt` to silence CMake warning.

**Comment by bradking** [comment]
> These variables are defined in src/CMakeLists.txt of this project at:

Ah, I had grepped just for `METAIO_INSTALL_BIN_DIR` and `METAIO_INSTALL_LIB_DIR`.  Those are not referenced.  They should be spelled `METAIO_INSTALL_RUNTIME_DIR` and `METAIO_INSTALL_LIBRARY_DIR`.

File: CMakeLists.txt [hunk 1]
```diff
@@ -71,4 +71,9 @@ else()
   option(BUILD_TESTING "Build the testing tree" ${BUILD_TESTING})
 endif()
 
+SET(METAIO_INSTALL_BIN_DIR ${ITK_INSTALL_BIN_DIR_CM24})
+SET(METAIO_INSTALL_LIB_DIR ${ITK_INSTALL_LIB_DIR_CM24})
+SET(METAIO_INSTALL_ARCHIVE_DIR ${METAIO_INSTALL_LIB_DIR})
+SET(METAIO_INSTALL_INCLUDE_DIR ${ITK_INSTALL_INCLUDE_DIR_CM24})
```
Comment (bradking) [inline CMakeLists.txt]: The first two should be spelled `METAIO_INSTALL_RUNTIME_DIR` and `METAIO_INSTALL_LIBRARY_DIR`.

Comment (schuhschuh) [reply CMakeLists.txt]: Oh, yes, good catch. Strange that previous patch did affect the location of the library, though, when I tested it.

File: CMakeLists.txt [hunk 2]
```diff
@@ -71,4 +71,10 @@ else()
   option(BUILD_TESTING "Build the testing tree" ${BUILD_TESTING})
 endif()
 
+SET(METAIO_INSTALL_EXPORT_NAME ${METAIO_NAMESPACE})
+SET(METAIO_INSTALL_RUNTIME_DIR ${ITK_INSTALL_BIN_DIR_CM24})
+SET(METAIO_INSTALL_LIBRARY_DIR ${ITK_INSTALL_LIB_DIR_CM24})
+SET(METAIO_INSTALL_ARCHIVE_DIR ${METAIO_INSTALL_LIB_DIR})
```
Comment (bradking) [inline CMakeLists.txt]: The reference to `METAIO_INSTALL_LIB_DIR` also needs to be updated.

**Comment by schuhschuh** [comment]
Ok, fixed up METAIO_INSTALL_ARCHIVE_DIR. Can also squash the commits (though I thought you may use GitHub's "Squash and Merge" feature instead).

When testing the changes again, I noticed now that when not the system zlib library should be used, the installation has another issue. The installation tree looks like:

```
/opt/metaio
├── include
│   └── itkMetaIO
│       ├── Utilities
│       │   └── itkzlib
│       │       ├── itk_zlib_mangle.h
│       │       ├── zconf.h
│       │       ├── zlib.h
│       │       └── zlibDllConfig.h
│       ├── itkThirdParty.h
│       ├── itk_zlib.h
│       ├── localMetaConfiguration.h
│       ├── metaArray.h
│       ├── metaArrow.h
│       ├── metaBlob.h
│       ├── metaCommand.h
│       ├── metaContour.h
│       ├── metaDTITube.h
│       ├── metaEllipse.h
│       ├── metaEvent.h
│       ├── metaFEMObject.h
│       ├── metaForm.h
│       ├── metaGaussian.h
│       ├── metaGroup.h
│       ├── metaIOConfig.h
│       ├── metaITKUtils.h
│       ├── metaImage.h
│       ├── metaImageTypes.h
│       ├── metaImageUtils.h
│       ├── metaLandmark.h
│       ├── metaLine.h
│       ├── metaMesh.h
│       ├── metaObject.h
│       ├── metaOutput.h
│       ├── metaScene.h
│       ├── metaSurface.h
│       ├── metaTransform.h
│       ├── metaTube.h
│       ├── metaTubeGraph.h
│       ├── metaTypes.h
│       ├── metaUtils.h
│       └── metaVesselTube.h
└── lib
    └── itkMetaIO
        ├── libITKMetaIO.a
        └── libitkzlib.a
```

But `itkMetaIO/itk_zlib.h` includes `itkzlib/zlib.h`, which won't be found if `itkMetaIO/Utilities` folder is not also added to the include search path. Any suggestion how to resolve this?

Could drop either drop the `Utilities` from the installation rule in `itkzlib/CMakeLists.txt`, or modify the `itk_zlib.h` in the top-level folder of this stand-alone MetaIO project to use include path `"Utilities/itkzlib/zlib.h"` instead. Any suggestions?

Can leave this for a separate PR as well...

**Comment by thewtex** [comment]
+1 for dropping *Utilities/*

**Comment by schuhschuh** [comment]
> +1 for dropping _Utilities/_

Done!

**Pull request merged**